core/controller/hook/jenkins.js

```diff
@@ -0,0 +1,59 @@
+/**
```

Comment:
so really all of Jenkins is being to be replaced by the socketio-powered slave system, right?